plugins/woocommerce/src/Internal/DataStores/Orders/LegacyDataHandler.php

```diff
@@ -322,8 +322,15 @@ public function get_order_from_datastore( int $order_id, string $data_store_id =
 		};
 		$update_data_store_func->call( $order, $data_store );
 
-		// Read order.
-		$data_store->read( $order );
+		// Read order (without triggering sync).
+		$prevent_sync_on_read = fn() => false;
```

Comment:
Sorry, I hadn't noticed this on my first review. Shouldn't this be just 'false'? [The filter is defined](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L1233-L1241) as accepting a bool argument.

Reply:
I'm not sure I follow @Konamiman. What do you mean? We can't pass `false` to `add_filter()` so that's why I'm wrapping it in a callback that _returns_ `false`. Am I missing something?

Reply:
Oh true, I got confused with `apply_filters`, sorry.

Still, this won't work: `remove_filter` doesn't work when the supplied target is a callback (and that's precisely why we use `AccessiblePrivateMethods` instead of callbacks that invoke these private methods); e.g. this will still print `false`:

```php
add_filter('foobar', fn() => false);
remove_filter('foobar', fn() => false);
echo var_export(apply_filters('foobar', true)) . "\n";
```

Instead you can use the built-in WordPress function `__return_false`:

```suggestion
		$prevent_sync_on_read = '__return_false';
```

and maybe you could even remove the `$prevent_sync_on_read` variable and inline the `__return_false` directly in `add_filter` and `remove_filter` (not mandatory of course, but that's how it's done in other places where this function is referenced in WooCommerce).

Reply:
Haha. No worries @Konamiman! I'm glad I wasn't crazy and that you also found your way to the truth! 😸 

> Still, using '__return_false' instead is probably cleaner, wdyt?

I deliberately chose not to use `__return_false` and instead add my own callback because users _might_ be hooking `__return_false` onto this filter already (in cases where they want to disable sync-on-read), and when we call `remove_filter()` it'll remove any instance that matches callback and priority, not just "the last one", so we might end up overriding the behavior users have configured. I'm not sure how likely it is that they'd use that same priority of 999 but it's still possible. With our own callback, though, we don't have that problem.

What do you think?

Reply:
> because users might be hooking __return_false onto this filter already

Ok, this makes sense... but I would add add a explicit comment about that. I (and I guess others too) would happily have refactored this to a `'__return_false'` had I encountered this code without further context.

Reply:
Addressed in e78d89ee5b60354e1837d6ab301d57277ed127be.